app/lib/search/index_simple.dart

```diff
@@ -155,35 +156,38 @@ class SimplePackageIndex implements PackageIndex {
   Map<String, double> getHealthScore(Iterable<String> packages) {
     return new Map.fromIterable(
       packages,
-      value: (String package) => (_packages[package].health ?? 0.0) * 100,
+      value: (package) => (_packages[package].health ?? 0.0),
     );
   }
 
   // visible for testing only
   Map<String, double> getPopularityScore(Iterable<String> packages) {
     return new Map.fromIterable(
       packages,
-      value: (String package) => _packages[package].popularity * 100,
+      value: (package) => _packages[package].popularity ?? 0.0,
     );
   }
 
   // visible for testing only
   Map<String, double> getMaintenanceScore(Iterable<String> packages) {
     return new Map.fromIterable(
       packages,
-      value: (String package) => (_packages[package].maintenance ?? 0.0) * 100,
+      value: (package) => (_packages[package].maintenance ?? 0.0),
     );
   }
 
+  Score _allPackages() =>
+      new Score(new Map.fromIterable(_packages.keys, value: (_) => 1.0));
```

Comment:
This would cause us to create a map of the size of `O(number-of-packages)` with values all being `1.0` - for every search. This is quite inefficient, given that the caller can simply use `1.0` if there is no score for it.

As mentioned on previous CLs, it is not very advisable to build up big data structures and then prune them later. Only build what you need.

Reply:
For now I'm not worried about this. Most queries I've measured are about 1-3 ms, and the maximum memory use of a search instance is between 380 and 420MB. I'd prefer to keep the code simpler in expense of memory use for now, and when we are happy with it, we can take a look in optimizations.